app/background-process/networks/dat/storage.js

```diff
@@ -0,0 +1,138 @@
+var path = require('path')
+var fs = require('fs')
+var raf = require('random-access-file')
+var multi = require('multi-random-access')
+var mkdirp = require('mkdirp')
+var rimraf = require('rimraf')
+var messages = require('append-tree/messages')
+var stat = require('hyperdrive/lib/messages').Stat
+
+/**
+ this is the internal storage policy for dats that's used in beaker
+ it's basically similar to hyperdrive's default
+ except:
+   1) content is stored as individual files, with the version # appended
+   2) if a `latestDir` is provided, the latest version is hardlinked to it
+**/
+
+export default function (metadataDir, latestDir) {
+  return {
+    metadata: function (name, opts) {
+      return raf(path.join(metadataDir, 'metadata', name))
+    },
+    content: function (name, opts, archive) {
+      if (!archive) archive = opts
+      if (name === 'data') return createStorage(archive, metadataDir, latestDir)
+      return raf(path.join(metadataDir, 'content', name))
+    }
+  }
+}
+
+function createStorage (archive, metadataDir, latestDir) {
+  return multi({limit: 128}, locate)
+
+  function locate (offset, cb) {
+    archive.ready(function (err) {
+      if (err) return cb(err)
+
+      find(archive.metadata, offset, function (err, node, st, index) {
+        if (err) return cb(err)
+        if (!node) return cb(new Error('Could not locate data'))
+
+        // provide the storage
+        var historicFilePath = path.join(metadataDir, 'content', 'files', `${node.name}.${st.offset}`)
+        cb(null, {
+          start: st.byteOffset,
+          end: st.byteOffset + st.size,
+          storage: raf(historicFilePath)
+        })
```

Comment:
my gut tells me we should wait on calling the callback until after the hardlinker has run ... for safety reasons